Riot/AppDelegate.m

```diff
@@ -1097,6 +1113,46 @@ - (void)application:(UIApplication *)application didRegisterUserNotificationSett
     }
 }
 
+- (void)application:(UIApplication *)application handleActionWithIdentifier:(NSString *)identifier forLocalNotification:(UILocalNotification *)notification withResponseInfo:(NSDictionary *)responseInfo completionHandler:(void (^)())completionHandler
+{
+    if ([identifier isEqualToString: @"inline-reply"])
+    {
+        NSString* roomId = notification.userInfo[@"room_id"];
+        if (roomId.length)
+        {
+            NSArray* mxAccounts = [MXKAccountManager sharedManager].activeAccounts;
+            MXRoom* room = nil;
+            for (MXKAccount* account in mxAccounts)
+            {
+                room = [account.mxSession roomWithRoomId:roomId];
+                if (room)
+                {
+                    break;
+                }
+            }
+            if (room == nil)
+            {
+                NSLog(@"[AppDelegate][Push] handleActionWithIdentifier: room with id %@ not found", roomId);
+            }
+            else
+            {
+                NSString* responseText = [responseInfo objectForKey:UIUserNotificationActionResponseTypedTextKey];
+                if (responseText != nil && responseText.length != 0)
+                {
+                    NSLog(@"[AppDelegate][Push] handleActionWithIdentifier: sending message to room: %@", roomId);
+                    [room sendTextMessage:responseText success:^(NSString* eventId) {} failure:^(NSError* error) {
+                        NSLog(@"[AppDelegate][Push] handleActionWithIdentifier: error sending text message: %@", error);
```

Comment:
How to deal better with the error? Is it possible to display it to the end user?
My concern is e2e rooms where sending can fail because there are new devices in a room. The sending fails because the user has not validated those new devices yet.

Reply:
Yeah I encountered that issue in my testing... Couldn’t figure out how to properly notify the user. How about another local notification that indicates there was an error sending the message?

Reply:
This is a good idea. The user should be able to go the room when clicking on this notification.

Then, the app will automatically propose them to resend the message. This failure flow seems not too bad.